src/tree/tree-item.js

```diff
@@ -0,0 +1,52 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import ActiveIdContext from './active-id-context.js';
+import OpenIdsContext from './open-ids-context.js';
+import combineRenderProps from '../utils/combine-render-props.js';
+
+export default function TreeItem({
+  children,
+  id,
+  title,
+}) {
+  const Consumer = combineRenderProps(
+    ActiveIdContext.Consumer,
+    OpenIdsContext.Consumer,
+  );
+
+  return (
+    <Consumer>
+      {(activeId, openIds) => {
```

Comment:
Note the flattened context consumers, instead of nesting them like:

```js
<FirstConsumer>
  {(props1) => (
    <SecondConsumer>
      {(props2) => (
```

Figured that flatter was easier to understand in this component, with the cost that the `combineRenderProps` function is mind-bending (basically an [uncurry](https://www.haskell.org/hoogle/?hoogle=uncurry) working on react components with children render-psops).